### ISSUE TITLE
validation_split

### ISSUE THREAD

**Issue by qinst64** [keras-team/keras#10571]
The description and code are not consistent:

1. [keras offical doc](https://keras.io/getting-started/faq/#how-is-the-validation-split-computed)
If you set the validation_split argument in model.fit to e.g. 0.1, then the validation data used will be __the last 10% of the data__. If you set it to 0.25, it will be the last 25% of the data, etc. Note that the data isn't shuffled before extracting the validation split, so the validation is literally just the last x% of samples in the input you passed.

2.  [keras  code](https://github.com/keras-team/keras-preprocessing/blob/master/keras_preprocessing/image.py)
about line 1394
``` python
split_idx = int(len(x) * image_data_generator._validation_split)
if subset == 'validation':
	x = x[:split_idx]
	x_misc = [np.asarray(xx[:split_idx]) for xx in x_misc]
	if y is not None:
		y = y[:split_idx]
else:
	x = x[split_idx:]
	x_misc = [np.asarray(xx[split_idx:]) for xx in x_misc]
	if y is not None:
		y = y[split_idx:]
```
so validation is __the fist x% of the data__.

**Comment by HughKu** [comment]
Hi @qinst64, maybe the following can help you.

`validation_split` in the doc you're referring to describes the `model.fit` behavior, which is defined in [Keras core repo](https://github.com/keras-team/keras/blob/454be50414967002197cc40be9d968a16a07f6b9/keras/engine/training.py#L981), not the one in keras-preprocessing module.

**Comment by qinst64** [comment]
@HughKu Thank you! I didn't notice it since I was using `fit_generator`. 
Now:
- `model.fit`  use  last x% of data
- `model.fit_generator` use first x% of data (behavior of `keras.preprocessing.image.ImageDataGenerator`)
 
Their differences make users confusion though. Guess better one of them (maybe the latter) to be changed in the future release.
